runtimes/pythonrt/runner_manager_remote_client.go

```diff
@@ -63,10 +169,116 @@ func configureRemoteRunnerManager(cfg RemoteRuntimeConfig) error {
 	return nil
 }
 
-func (rrm *remoteRunnerManager) Start(ctx context.Context, sessionID sdktypes.SessionID, buildArtifacts []byte, vars map[string]string) (string, *RunnerClient, error) {
-	// rrm.remoteManagers[0].StartRunner(ctx, &pb.StartRunnerRequest{})
-	return "", nil, nil
+func createRunnerAuthInterceptor(token string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		mdCtx := metadata.NewOutgoingContext(ctx, metadata.Pairs("token", token))
+		return invoker(mdCtx, method, req, reply, cc, opts...)
+	}
+}
+
+func (rrm *remoteRunnersManager) Start(ctx context.Context, sessionID sdktypes.SessionID, buildArtifacts []byte, vars map[string]string) (string, *RunnerClient, error) {
+	runnerManager, err := rrm.acquireRemoteManager()
+	if err != nil {
+		return "", nil, err
+	}
+
+	rrm.logger.Info(fmt.Sprintf("using runner manager %s", runnerManager.addr))
+
+	userCodePath, err := prepareUserCode(buildArtifacts, false)
+	if err != nil {
+		return "", nil, err
+	}
+
+	t := tar.NewTarFile()
+
+	filepath.Walk(userCodePath, func(path string, info os.FileInfo, err error) error {
```

Comment:
See GH analysis on checking `Walk` error. Also https://pkg.go.dev/archive/tar#Writer.AddFS
And also move to a utility function. I think we already have one?
Also, also, use the built-in tar with an underlying `bytes.Buffer`